src/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -157,7 +255,545 @@ public void ContinueTimeout_SetNegativeOne_Success(Uri remoteServer)
         public void ContinueTimeout_SetNegativeTwo_ThrowsArgumentOutOfRangeException(Uri remoteServer)
         {
             HttpWebRequest request = WebRequest.CreateHttp(remoteServer);
-            Assert.Throws<ArgumentOutOfRangeException>(() => request.ContinueTimeout = -2);
+            Assert.Throws<ArgumentOutOfRangeException>("value", () => request.ContinueTimeout = -2);
+        }
+
+        [Theory, MemberData(nameof(EchoServers))]
+        public void Timeout_SetThenGetZero_ExpectZero(Uri remoteServer)
+        {
+            const int timeout = 0;
+            HttpWebRequest request = WebRequest.CreateHttp(remoteServer);
+            request.Timeout = timeout;
+            Assert.Equal(timeout, request.Timeout);
+        }
+
+        [Theory, MemberData(nameof(EchoServers))]
+        public void Timeout_SetNegativeOne_Success(Uri remoteServer)
+        {
+            HttpWebRequest request = WebRequest.CreateHttp(remoteServer);
+            request.Timeout = -1;
+        }
+
+        [Theory, MemberData(nameof(EchoServers))]
+        public void Timeout_SetNegativeTwo_ThrowsArgumentOutOfRangeException(Uri remoteServer)
+        {
+            HttpWebRequest request = WebRequest.CreateHttp(remoteServer);
+            Assert.Throws<ArgumentOutOfRangeException>("value", () => request.Timeout = -2);
+        }
+
+        [Theory, MemberData(nameof(EchoServers))]
+        public void TimeOut_SetThenGet_ValuesMatch(Uri remoteServer)
+        {
+            HttpWebRequest request = WebRequest.CreateHttp(remoteServer);
+            request.Timeout = 100;
+            Assert.Equal(100, request.Timeout);
+
+            request.Timeout = Threading.Timeout.Infinite;
+            Assert.Equal(Threading.Timeout.Infinite, request.Timeout);
+
+            request.Timeout = int.MaxValue;
+            Assert.Equal(int.MaxValue, request.Timeout);
+        }
+
+        [Fact]
+        public async Task Timeout_SetTenMillisecondsOnLoopback_ThrowsWebException()
+        {
+            await LoopbackServer.CreateServerAsync((server, url) =>
+            {
+                HttpWebRequest request = WebRequest.CreateHttp(url);
+                request.Timeout = 10; // ms.
+
+                var sw = Stopwatch.StartNew();
+                WebException exception = Assert.Throws<WebException>(() =>
+                {
+                    var response = (HttpWebResponse)request.GetResponse();
+                });
+                
+                sw.Stop();
+
+                Assert.InRange(sw.ElapsedMilliseconds, 1, 5 * 1000);
```

Comment:
It was set to that value because an external remote endpoint was used. Now it uses a local loopback server